assets/js/base/hooks/payment-methods/use-payment-events.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useState, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STATUS } from './constants';
```

Comment:
This is pretty minor, but I wonder if grouping all the statuses into one constant and then referencing them using property access is really that helpful since one of the major benefits of using constants is to prevent typos. Below I could write `STATUS.FALIED`, which would happily evaluate to `undefined` and it might be hard bug to locate.

It's not perfect, but one way to help this might be to dereference all the statuses here and use them without dot notation. That way there's at least only one place where a typo could occur.

```js
const { PRISTINE, ERROR } = STATUS; // etc.
```

Probably a more robust technique would be to export and import each status itself, which would give a much more helpful error if there was a typo anywhere in this file.

```js
import { STATUS_PRISTINE, STATUS_ERROR } from './constants'; // etc.
```

Reply:
🤔 hmm good thoughts.  Destructuring would definitely help.  However in practice, I've noticed I get autocomplete for properties just as well as autocomplete for declared vars so it would have to be pretty intentional to do a typo for a non-existent property?

Reply:
For sure, editors with autocomplete or static analysis can take care of this sort of thing themselves, but I never like to assume that someone is using a particular kind of IDE (or that it is working).